library/std/src/sys/pal/unix/stack_overflow.rs

```diff
@@ -370,7 +372,8 @@ mod imp {
             // this way someone on any unix-y OS can check that all these compile
             if cfg!(all(target_os = "linux", not(target_env = "musl"))) {
                 install_main_guard_linux(page_size)
-            } else if cfg!(all(target_os = "linux", target_env = "musl")) {
+            } else if cfg!(any(all(target_os = "linux", target_env = "musl"), target_os = "cygwin"))
+            {
                 install_main_guard_linux_musl(page_size)
```

Comment:
Is this really the correct variant? If I understand correctly, Cygwin sets up its own main thread stack and correctly reports the limits of that stack in `pthread_attr_getstack` – so it should be possible to correctly report stack overflows.

Reply:
I will investigate it later.

Reply:
I tested some code and find that cygwin programs fail before the SIGSEGV handler is called. So I removed the code here. I choose to just not set the stack overflow handler here.